### ISSUE TITLE
Events Pokemon RNG

### ISSUE THREAD

**Issue by Real96** [wwwwwwzx/SMEncounterRNGTool#16]
Events pokemon need to be tested. Pokemon with fixed stats and without fixed stats.
A good place to RNG them is the first Pokemon Center you encounter during the storyline

**Comment by wwwwwwzx** [comment]
Does that PC have the least number of NPCs?

**Comment by Real96** [comment]
Mhmhmh honestly i didn't made really a big research xD I'm doing it now, i'll finish in a bit and i will tell you if there are any PC with less NPCs

**Comment by Real96** [comment]
If also pokemon are considered as NPCs, this is the only PC with 8 NPCs. The other are 9+

**Comment by wwwwwwzx** [comment]
Oh, that's challenging, I have to find the timing of Pokemon determination, in my memory there is no renaming screen.

**Comment by Real96** [comment]
The rng is really similar to Type:Null. Sometimes i got -1 sometimes -5

**Comment by wwwwwwzx** [comment]
-1 or -5 In `show results after delay` mode?

**Comment by Real96** [comment]
Yes, i'm making more tests

**Comment by Real96** [comment]
I can't make the frame follow the Timeline....did you figured out something?

**Comment by wwwwwwzx** [comment]
Well I left the game running for 4 mins. It follows timeline perfectly. And I RNG another shiny Type:null in one try.

**Comment by wwwwwwzx** [comment]
Be sure to **FIX** the starting Frame Range at the `safe frame` **w/o** `createtimeline`

**Comment by wwwwwwzx** [comment]
Wait, I'm also in trouble...

**Comment by wwwwwwzx** [comment]
Machamp was solved, just use Time delay = 4, NPC =6. All locked except IVs. Working on munchlax

**Comment by Real96** [comment]
Great! Also there is Sapporo Vulpix (nothing fixed) and Pokémon with 3 stats at 31, but still nothing has been released

**Comment by wwwwwwzx** [comment]
When you stand behind the delivery man, npc=4. but still problematic

**Comment by Real96** [comment]
Yes i tryed to stand behing, but sometimes the number of the npc seems to change....

**Comment by wwwwwwzx** [comment]
I guess there are two types of events Pokemon, one is in your trainer ID, you have the chance to change its name, there is a long delay.  Another is fixed ID, which will be determined almost right away(4F) and it's much easier

**Comment by wwwwwwzx** [comment]
A rough result `Time Delay `for munchlax is 70. Only Check the `Ablity Locked` But it's really weird

**Comment by Real96** [comment]
Did you hit some framse?

**Comment by wwwwwwzx** [comment]
Machamp is easy to hit, if it's not following the timeline just change the npc to advancement - 1, since the delay is only 4 "Select" Pressing. I guess most of events Pokemon will be this type.

**Comment by Real96** [comment]
Ok, but do i need to arrive to one of the frame right? I tryed munchlax, but i got -6. Delay 70 NPCs 6

**Comment by wwwwwwzx** [comment]
I figured out. update soon

**Comment by Real96** [comment]
Really? :O Great! ANyway machamp has got a fixed 31, do i need to put Atk 31 ~ 31?

**Comment by wwwwwwzx** [comment]
Do you see the setting box in the second tab? Check the checkbox after Atk, and set 31 in the box between.
I just update my model,` Time Delay = 62` , NPC= 4 or 6 both work. Check `Your ID` and `Ability Locked` for munchlax
I'm going to bed now. Hope it's correct

**Comment by Real96** [comment]
Fantastic! I'll make some tests, see you later and good night! And thanks a lot!

**Comment by Real96** [comment]
Just tried to RNG Machamp, Sapporo Vulpix and Munchlax. I didn't have any problems! It worked perfectly! :D
I needed to select `Ability Locked` for all of them to make the reseach show correct natures (most of the events has got Ability locked). The frame followed the `Timeline` perfectly, with NPCs 6.

I also tried Lunar Shiny Magikarp, but there is a litte "bug" which tell you that PSV and PID are different from PSV and PID of the pokemon you get. Anyway all the other stats are correct. I'm opening an issue for the Always Shiny option.
